File: Python/python/Abstract_Factory/SQLUserDAO.py
from __future__ import annotations
from typing import List
from .UserDAO import UserDAO
from .TransferObject import TransferObject
from .SQLDataSource import SQLDataSource

# 4. Implementación concreta del Producto
class SQLUserDAO(UserDAO):
    def __init__(self):
        self.data_source = SQLDataSource()

    def get_users(self) -> List[TransferObject]:
        raw_data = self.data_source.get_data()
        return [TransferObject(data) for data in raw_data]

File: src/abstract_Factory/SQLUserDAO.java
package abstract_Factory;

import java.util.List;
import java.util.stream.Collectors;


public class SQLUserDAO implements UserDAO{
    private SQLDataSource dataSource = new SQLDataSource();

    @Override
    public List<TransferObject> getUsers() {
        List<String> rawData = dataSource.getData();
        return rawData.stream()
                .map(TransferObject::new)
                .collect(Collectors.toList());
    }
}
